File: mpp-proiect-repository-cristinahendre/lab12/lab6/modul/DomainModule/src/main/java/Domain/Donator.java
package Domain;

import java.io.Serializable;
import java.util.List;
import java.util.Objects;

public class Donator extends  Entity<Integer> implements Serializable {


    private String nume;
    private String prenume;
    private String adresa;
    private long nrTelefon;

    public Donator(){}

    public String getNume() {
        return nume;
    }

    public void setNume(String nume) {
        this.nume = nume;
    }

    public String getPrenume() {
        return prenume;
    }

    public void setPrenume(String prenume) {
        this.prenume = prenume;
    }

    public String getAdresa() {
        return adresa;
    }

    public void setAdresa(String adresa) {
        this.adresa = adresa;
    }

    public long getNrTelefon() {
        return nrTelefon;
    }

    public void setNrTelefon(long nrTelefon) {
        this.nrTelefon = nrTelefon;
    }

    public String getAdresaSpatii(){
        String[] list = adresa.split(",");
        String rez="";
        for(String cuv: list){
            rez+=cuv;
            rez+='\n';
        }
        return rez;

    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Donator)) return false;
        Donator donator = (Donator) o;
        return nrTelefon == donator.nrTelefon && nume.equals(donator.nume) && prenume.equals(donator.prenume) && adresa.equals(donator.adresa);
    }

    @Override
    public int hashCode() {
        return Objects.hash(nume, prenume, adresa, nrTelefon);
    }

    @Override
    public String toString() {
        return "Donator{" +
                " id = "+this.getId()+ " "+
                "nume='" + nume + '\'' +
                ", prenume='" + prenume + '\'' +
                ", adresa='" + adresa + '\'' +
                ", nrTelefon=" + nrTelefon +
                '}';
    }

    public Donator(String nume, String prenume, String adresa, long nrTelefon) {
        this.nume = nume;
        this.prenume = prenume;
        this.adresa = adresa;
        this.nrTelefon = nrTelefon;
    }


}

File: mpp-proiect-repository-cristinahendre/C#/Cerinta1/Cerinta1/Domain/Donator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Cerinta1.Domain
{
    class Donator: Entity<int>
    {
        string nume;
        string prenume;
        string adresa;
        long nrTelefon;

        public Donator(string nume, string prenume, string adresa, long nrTelefon)
        {
            this.nume = nume;
            this.prenume = prenume;
            this.adresa = adresa;
            this.nrTelefon = nrTelefon;
        }

        public string Nume { get => nume; set => nume = value; }
        public string Prenume { get => prenume; set => prenume = value; }
        public string Adresa { get => adresa; set => adresa = value; }
        public long NrTelefon { get => nrTelefon; set => nrTelefon = value; }

        public override bool Equals(object obj)
        {
            return obj is Donator donator &&
                   nume == donator.nume &&
                   prenume == donator.prenume &&
                   adresa == donator.adresa &&
                   nrTelefon == donator.nrTelefon;
        }

        public override int GetHashCode()
        {
            int hashCode = -2092692272;
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(nume);
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(prenume);
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(adresa);
            hashCode = hashCode * -1521134295 + nrTelefon.GetHashCode();
            return hashCode;
        }

        public override string ToString()
        {
            return "Id = " + this.Id + "  ,nume = " + this.GetFullName() + " , adresa = " +
                this.Adresa + "  , nr. telefon = " + this.NrTelefon;
        }

        public String GetFullName()
        {
            return this.Nume + " " + this.Prenume ;
        }
    }
}
